### PR TITLE
[path_provider] Revert Gradle and androidx.annotation bump in path_provider_android

### PR DESCRIPTION
**Pull request by camsim99**
Reverts https://github.com/flutter/plugins/pull/6189 and https://github.com/flutter/plugins/pull/6228 to fix https://github.com/flutter/flutter/issues/109397.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/path_provider/path_provider_android/pubspec.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: path_provider_android
 description: Android implementation of the path_provider plugin.
 repository: https://github.com/flutter/plugins/tree/main/packages/path_provider/path_provider_android
 issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+path_provider%22
-version: 2.0.19
+version: 2.0.18
```
Comment (christopherfujino) [inline packages/path_provider/path_provider_android/pubspec.yaml:5]: since we already published 2.0.19, I think we'll have to name this 2.0.20

**Comment by christopherfujino** [comment]
should we also roll back Gary's Kotlin change in https://github.com/flutter/plugins/pull/6228 ?

**Comment by camsim99** [comment]
> should we also roll back Gary's Kotlin change in #6228 ?

I asked @GaryQian and he thought it would be safe to keep it. I don't have much context on the impact of the change, so I'd be fine reverting it just in case.

**Comment by GaryQian** [comment]
You can roll kotlin back too just in case. It's super easy to bump it again, let's green the tree first

**Comment by christopherfujino** [comment]
> You can roll kotlin back too just in case. It's super easy to bump it again, let's green the tree first

The tree is green now because I pinned path_provider_android to 2.0.17. If we can validate this change works on the affected devicelab build, we can un-pin the package and the bot will roll it forward.

**Review by GaryQian — APPROVED** [review]
LGTM!

**Pull request merged**